File: 3633_Earliest_Finish_Time_for_Land_and_Water_Rides_I/solution.c
#define MAX(a, b) ((a) > (b) ? (a) : (b))
#define MIN(a, b) ((a) < (b) ? (a) : (b))
int earliestFinishTime(int* lStartTime, int n, int* lDuration, int lDurationSize, int* wStartTime, int m, int* wDuration, int wDurationSize) {
    int r = INT_MAX;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            int l = lStartTime[i] + lDuration[i];
            int lw = MAX(l, wStartTime[j]) + wDuration[j];
            r = MIN(r, lw);

            int w = wStartTime[j] + wDuration[j];
            int wl = MAX(w, lStartTime[i]) + lDuration[i];
            r = MIN(r, wl);
        }
    }
    return r;
}

File: 3633_Earliest_Finish_Time_for_Land_and_Water_Rides_I/solution.py
class Solution:
    def earliestFinishTime(self, lStartTime: List[int], lDuration: List[int], wStartTime: List[int], wDuration: List[int]) -> int:
        n = len(lStartTime)
        m = len(wStartTime)
        r = inf
        for i in range(n):
            for j in range(m):
                l = lStartTime[i] + lDuration[i]
                lw = max(l, wStartTime[j]) + wDuration[j]
                r = min(r, lw)

                w = wStartTime[j] + wDuration[j]
                wl = max(w, lStartTime[i]) + lDuration[i]
                r = min(r, wl)
        return r
